model/riscv_sys_regs.sail

```diff
@@ -566,6 +574,40 @@ function retire_instruction() -> unit = {
   if minstret_increment then minstret = minstret + 1;
 }
 
+// HPM (Hardware Performance Monitoring) counters. The lowest three values are
+// not used but they are defined for simplicity.
+register mhpmcounter : vector(32, dec, bits(64))
+
+// HPM events selector. These control what the HPM counters measure. The lowest
+// three values are not used but they are defined for simplicity.
+register mhpmevent : vector(32, dec, xlenbits)
+
+// Valid HPM counter indices. The lowest three are used for mcycle, mtime and minstret.
+type hpmidx = range(3, 31)
+
+// Convert the lowest 5 bits of a CSR index to an hpmidx. Asserts if it is 0..2.
+function hpmidx_from_bits(b : bits(5)) -> hpmidx = {
+  let index = unsigned(b);
+  if index >= 3 then index else {
+    assert(false, "unreachable HPM index");
+    undefined
+  }
+}
```

Comment:
```suggestion
function hpmidx_from_bits(b : bits(5)) -> hpmidx = {
  assert(unsigned(b) >= 3, "unreachable HPM index");
  unsigned(b)
}
```

Does this type check?

Reply:
It doesn't, but I updated it to this which does:

```
  let index = unsigned(b);
  assert(index >= 3, "unreachable HPM index");
  index
  ```
  
Not sure what I was thinking :-D